src/proposals/20240110/SparkEthereum_20240110.t.sol

```diff
@@ -0,0 +1,637 @@
+        // 4. Revert to reset state to before claim
+
+        vm.revertTo(snapshot);
+
+        // 5. Supply 10% of more tokens
+
+        vm.startPrank(WHALE1);
+
+        uint256 amount = wethAToken.balanceOf(WHALE1) / 10;
+
+        deal(WETH, WHALE1, amount);
+
+        IERC20(WETH).approve(POOL, amount);
+        IPool(POOL).supply(WETH, amount, WHALE1, 0);
+
+        vm.stopPrank();
+
+        // 6. Claim rewards for whale 1 (with supply)
+
+        vm.prank(WHALE1);
+        incentivesController.claimAllRewards(assets, claimAddress1);
+
+        // 7. Assert that the second claim are is same as before, meaning the mint updated the
+        //    accrued rewards for the user.
+
+        assertEq(wsteth.balanceOf(claimAddress1), firstClaimAmount);
+    }
+
+    function test_claimAllRewards_withdrawAfterWarp() external {
+        _setUpRewards();
+
+        address[] memory assets = new address[](1);
+        assets[0] = WETH_ATOKEN;
+
+        vm.startPrank(WHALE1);
+
+        // 1. Warp halfway through the rewards period
+
+        skip(DURATION / 2);
+
+        // 2. Snapshot state
+
+        uint256 snapshot = vm.snapshot();
+
+        // 3. Claim rewards for whale 1 (without withdraw)
+
+        incentivesController.claimAllRewards(assets, claimAddress1);
+
+        uint256 firstClaimAmount = wsteth.balanceOf(claimAddress1);
+
+        // 4. Revert to reset state to before claim
+
+        vm.revertTo(snapshot);
+
+        // 5. Withdraw 10% of tokens
+
+        uint256 amount = wethAToken.balanceOf(WHALE1) / 10;
+
+        IPool(POOL).withdraw(WETH, amount, WHALE1);
+
+        // 6. Claim rewards for whale 1 (with withdraw)
+
+        incentivesController.claimAllRewards(assets, claimAddress1);
+
+        // 7. Assert that the second claim are is same as before, meaning the burn updated the
+        //    accrued rewards for the user.
+
+        assertEq(wsteth.balanceOf(claimAddress1), firstClaimAmount);
+    }
+
+    function _getTotalExpectedRewards(address user) internal view returns (uint256 expectedTotalRewards) {
+        expectedTotalRewards = _getExpectedRewards(user, DURATION);
+    }
+
+    function _getExpectedRewards(address user, uint256 earningDuration)
+        internal view returns (uint256 expectedRewards)
+    {
+        uint256 rewardShare = wethAToken.scaledBalanceOf(user) * 1e18 / wethAToken.scaledTotalSupply();
+
+        expectedRewards = REWARD_AMOUNT * rewardShare * earningDuration / 1e18 / DURATION;
+    }
+
+    function assertFrozen(string memory assetSymbol, bool frozen) internal {
+        assertEq(_findReserveConfigBySymbol(createConfigurationSnapshot('', pool), assetSymbol).isFrozen, frozen);
+    }
+
+    function assertPaused(string memory assetSymbol, bool paused) internal {
+        assertEq(_findReserveConfigBySymbol(createConfigurationSnapshot('', pool), assetSymbol).isPaused, paused);
+    }
+
+    function test_freezerMomE2E() public {
+        GovHelpers.executePayload(vm, payload, executor);
```

Comment:
NOT BLOCKING DEPLOYMENT - Lets update this test to use all the real deployed freeze/pause spells and give each of them the hat. I would also just make each its own test so we're working against clean state each time.

Reply:
this is done now btw can close